test/caption-stream.test.js

```diff
@@ -671,38 +675,54 @@ QUnit.test('drops data until first command that sets activeChannel for a field',
   });
 
   var packets = [
+    // test that packets in same field and same data channel are dropped
+    // before a control code that sets the data channel
     { pts: 0 * 1000, ccData: characters('no'), type: 0 },
     { pts: 0 * 1000, ccData: characters('t '), type: 0 },
     { pts: 0 * 1000, ccData: characters('th'), type: 0 },
     { pts: 0 * 1000, ccData: characters('is'), type: 0 },
+    // EOC (end of caption), sets CC1
     { pts: 1 * 1000, ccData: 0x142f, type: 0 },
+    // RCL (resume caption loading)
     { pts: 1 * 1000, ccData: 0x1420, type: 0 },
+    // EOC, if data wasn't dropped this would dispatch a caption
     { pts: 2 * 1000, ccData: 0x142f, type: 0 },
-    // RCL, resume caption loading
+    // RCL
     { pts: 3 * 1000, ccData: 0x1420, type: 0 },
-    { pts: 3 * 1000, ccData: 0x142e, type: 0 },
     { pts: 4 * 1000, ccData: characters('fi'), type: 0 },
     { pts: 4 * 1000, ccData: characters('el'), type: 0 },
     { pts: 4 * 1000, ccData: characters('d0'), type: 0 },
+    // EOC, mark display start
+    { pts: 5 * 1000, ccData: 0x142f, type: 0 },
+    // EOC, duplicated as per spec
     { pts: 5 * 1000, ccData: 0x142f, type: 0 },
-    { pts: 5 * 1000, ccData: 0x1420, type: 0 },
+    // EOC, mark display end and flush
     { pts: 6 * 1000, ccData: 0x142f, type: 0 },
+    // EOC not duplicated cuz not necessary
```

Comment:
`cuz`? 🙃 Is the reason it's not necessary because it's the 3rd EOC?

Reply:
It's not necessary because duplicate control codes are simply ignored (as opposed to waiting for the duplicated control code to process it). We only need to duplicate a control code if it's used twice in a row (e.g. to flush a pop-on caption). I could duplicate all control codes always in all tests, but I feel like that would be overkill.

Reply:
Oh yeah, I see. I agree the comment is probably enough